stdlib/src/collections/dict.mojo

```diff
@@ -78,18 +80,33 @@ struct _DictEntryIter[
     @always_inline
     fn __next__(inout self) -> Self.ref_type:
         while True:
-            debug_assert(self.index < self.src[]._reserved, "dict iter bounds")
+            if forward:
```

Comment:
**Suggestion** Since `forward` is a parameter and not a runtime value (as it should be), we can evaluate this branch at compile-time using `@parameter if` instead of `if`.  Ditto below.